sdks/python/apache_beam/coders/avro_generic_coder.py

```diff
@@ -0,0 +1,99 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""avro generic record coder for simple dict-type x-language data transfer."""
+
+from __future__ import absolute_import
+
+import json
+from io import BytesIO
+
+from fastavro import parse_schema
+from fastavro import schemaless_reader
+from fastavro import schemaless_writer
+
+from apache_beam.coders.coder_impl import SimpleCoderImpl
+from apache_beam.coders.coders import Coder
+from apache_beam.coders.coders import FastCoder
+
+AVRO_GENERIC_CODER_URN = "beam:coder:avro:generic:v1"
+
+__all__ = ['AvroGenericCoder', 'AvroGenericRecord']
+
+
+class AvroGenericCoder(FastCoder):
+  """A coder used for AvroGenericRecord values."""
+
+  def __init__(self, schema):
+    self.schema = schema
+
+  def _create_impl(self):
+    return AvroGenericCoderImpl(self.schema)
+
+  def is_deterministic(self):
+    # TODO: need to confirm if it's deterministic
+    return False
```

Comment:
You have probably seen `AvroDeterminismChecker`. So false seems safe here if we do not have a checker in Python.

Reply:
https://issues.apache.org/jira/browse/BEAM-7308